gobblin-runtime/src/main/java/org/apache/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -679,6 +694,7 @@ public void apply(JobListener jobListener, JobContext jobContext)
             }
           }
         }
+        jobStoppedGracefully = true;
```

Comment:
We should set this in final block?

Reply:
Yes, this means that the finally got to execute in a normal way, so shutdown hook should not double try to close again.
Did that answer your comment?